Archive failures when included in framework built with BUILD_LIBRARY_FOR_DISTRIBUTION=YES

My framework fails to archive successfully when this library is included and I have the library evolution flag of `BUILD_LIBRARY_FOR_DISTRIBUTION=YES`

Is this expected? Is it desired to have this library patched such that library evolution archives succeed? Specifically, I get the following errors:

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncChunkedByGroupSequence.swift:108:10: error: 'let' property 'base' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.base = base
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncChunkedByGroupSequence.swift:109:10: error: 'let' property 'grouping' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.grouping = grouping
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncChunkedOnProjectionSequence.swift:89:10: error: 'let' property 'base' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.base = base
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncChunkedOnProjectionSequence.swift:90:10: error: 'let' property 'projection' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.projection = projection
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncChunksOfCountSequence.swift:74:10: error: 'let' property 'base' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.base = base
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncChunksOfCountSequence.swift:75:10: error: 'let' property 'count' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.count = count
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncInterspersedSequence.swift:38:10: error: 'let' property 'base' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.base = base
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncInterspersedSequence.swift:39:10: error: 'let' property 'separator' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.separator = separator
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncInterspersedSequence.swift:67:12: error: 'let' property 'separator' may not be initialized directly; use "self.init(...)" or "self = ..." instead
      self.separator = separator
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncJoinedBySeparatorSequence.swift:133:10: error: 'let' property 'base' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.base = base
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncJoinedBySeparatorSequence.swift:134:10: error: 'let' property 'separator' may not be initialized directly; use "self.init(...)" or "self = ..." instead
    self.separator = separator
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncRemoveDuplicatesSequence.swift:52:12: error: 'let' property 'predicate' may not be initialized directly; use "self.init(...)" or "self = ..." instead
      self.predicate = predicate
```

```
swift-async-algorithms/Sources/AsyncAlgorithms/AsyncRemoveDuplicatesSequence.swift:111:12: error: 'let' property 'predicate' may not be initialized directly; use "self.init(...)" or "self = ..." instead
      self.predicate = predicate
```

--- a/Sources/AsyncAlgorithms/AsyncChunkedByGroupSequence.swift
+++ b/Sources/AsyncAlgorithms/AsyncChunkedByGroupSequence.swift
@@ -103,7 +103,7 @@ public struct AsyncChunkedByGroupSequence<Base: AsyncSequence, Collected: RangeR
   @usableFromInline
   let grouping : @Sendable (Base.Element, Base.Element) -> Bool
 
-  @inlinable
+  @usableFromInline
   init(_ base: Base, grouping: @escaping @Sendable (Base.Element, Base.Element) -> Bool) {
     self.base = base
     self.grouping = grouping

--- a/Sources/AsyncAlgorithms/AsyncChunkedOnProjectionSequence.swift
+++ b/Sources/AsyncAlgorithms/AsyncChunkedOnProjectionSequence.swift
@@ -84,7 +84,7 @@ public struct AsyncChunkedOnProjectionSequence<Base: AsyncSequence, Subject: Equ
   @usableFromInline
   let projection : @Sendable (Base.Element) -> Subject
 
-  @inlinable
+  @usableFromInline
   init(_ base: Base, projection: @escaping @Sendable (Base.Element) -> Subject) {
     self.base = base
     self.projection = projection
@@ -98,4 +98,3 @@ public struct AsyncChunkedOnProjectionSequence<Base: AsyncSequence, Subject: Equ
 
 extension AsyncChunkedOnProjectionSequence : Sendable where Base : Sendable, Base.Element : Sendable { }
 extension AsyncChunkedOnProjectionSequence.Iterator : Sendable where Base.AsyncIterator : Sendable, Base.Element : Sendable, Subject : Sendable { }
-

--- a/Sources/AsyncAlgorithms/AsyncChunksOfCountSequence.swift
+++ b/Sources/AsyncAlgorithms/AsyncChunksOfCountSequence.swift
@@ -68,7 +68,7 @@ public struct AsyncChunksOfCountSequence<Base: AsyncSequence, Collected: RangeRe
   @usableFromInline
   let count : Int
 
-  @inlinable
+  @usableFromInline
   init(_ base: Base, count: Int) {
     precondition(count > 0)
     self.base = base

--- a/Sources/AsyncAlgorithms/AsyncInterspersedSequence.swift
+++ b/Sources/AsyncAlgorithms/AsyncInterspersedSequence.swift
@@ -33,7 +33,7 @@ public struct AsyncInterspersedSequence<Base: AsyncSequence> {
   @usableFromInline
   internal let separator: Base.Element
 
-  @inlinable
+  @usableFromInline
   internal init(_ base: Base, separator: Base.Element) {
     self.base = base
     self.separator = separator
@@ -61,7 +61,7 @@ extension AsyncInterspersedSequence: AsyncSequence {
     @usableFromInline
     internal var state = State.start
 
-    @inlinable
+    @usableFromInline
     internal init(_ iterator: Base.AsyncIterator, separator: Base.Element) {
       self.iterator = iterator
       self.separator = separator

--- a/Sources/AsyncAlgorithms/AsyncJoinedBySeparatorSequence.swift
+++ b/Sources/AsyncAlgorithms/AsyncJoinedBySeparatorSequence.swift
@@ -74,7 +74,7 @@ public struct AsyncJoinedBySeparatorSequence<Base: AsyncSequence, Separator: Asy
     @usableFromInline
     var state: State
 
-    @inlinable
+    @usableFromInline
     init(_ iterator: Base.AsyncIterator, separator: Separator) {
       state = .initial(iterator, separator)
     }
@@ -128,7 +128,7 @@ public struct AsyncJoinedBySeparatorSequence<Base: AsyncSequence, Separator: Asy
   @usableFromInline
   let separator: Separator
 
-  @inlinable
+  @usableFromInline
   init(_ base: Base, separator: Separator) {
     self.base = base
     self.separator = separator

--- a/Sources/AsyncAlgorithms/AsyncJoinedSequence.swift
+++ b/Sources/AsyncAlgorithms/AsyncJoinedSequence.swift
@@ -79,7 +79,7 @@ public struct AsyncJoinedSequence<Base: AsyncSequence>: AsyncSequence where Base
   @usableFromInline
   let base: Base
   
-  @inlinable
+  @usableFromInline
   init(_ base: Base) {
     self.base = base
   }

--- a/Sources/AsyncAlgorithms/AsyncRemoveDuplicatesSequence.swift
+++ b/Sources/AsyncAlgorithms/AsyncRemoveDuplicatesSequence.swift
@@ -46,7 +46,7 @@ public struct AsyncRemoveDuplicatesSequence<Base: AsyncSequence>: AsyncSequence 
     @usableFromInline
     var last: Element?
 
-    @inlinable
+    @usableFromInline
     init(iterator: Base.AsyncIterator, predicate: @escaping @Sendable (Element, Element) async -> Bool) {
       self.iterator = iterator
       self.predicate = predicate
@@ -105,7 +105,7 @@ public struct AsyncThrowingRemoveDuplicatesSequence<Base: AsyncSequence>: AsyncS
     @usableFromInline
     var last: Element?
 
-    @inlinable
+    @usableFromInline
     init(iterator: Base.AsyncIterator, predicate: @escaping @Sendable (Element, Element) async throws -> Bool) {
       self.iterator = iterator
       self.predicate = predicate

